flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/BarrierBufferTestBase.java

```diff
@@ -531,7 +532,7 @@ public void testMultiChannelSkippingCheckpoints() throws Exception {
 
 		// checkpoint 3 aborted (end of partition)
 		check(sequence[20], buffer.getNextNonBlocked(), PAGE_SIZE);
-		verify(toNotify).abortCheckpointOnBarrier(eq(3L), any(CheckpointDeclineSubsumedException.class));
+		verify(toNotify).abortCheckpointOnBarrier(eq(3L), any(InputEndOfStreamException.class));
```

Comment:
@NicoK @zhijiangW I'd love to get your input here. As far as i can tell the existing code was simple wrong, as the supposed exception wasn't supposed to be thrown in this situation.

Reply:
alright so we need a matcher that actually checks the class?

Reply:
I supposed we should use `isA` instead.